File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/NotNullRule.java
package com.wesleyelliott.kubwa.rule;

import android.text.TextUtils;

/**
 * Created by wesley on 2016/07/28.
 */

public class NotNullRule extends Rule<String> {

    public NotNullRule() {
        super(String.class);
    }

    @Override
    public boolean isValid(String value) {
        return !TextUtils.isEmpty(value);
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/NotNullRule.kt
package com.wesleyelliott.kubwa.rule

import android.text.TextUtils

/**
 * Created by wesley on 2016/07/28.
 */

class NotNullRule : Rule<String>(String::class.java) {

    override fun isValid(value: String?): Boolean {
        return !TextUtils.isEmpty(value)
    }
}
